auth-external-api/src/test/java/uk/gov/di/authentication/external/lambda/UserInfoHandlerTest.java

```diff
@@ -77,18 +85,22 @@ void shouldReturn200WithUserInfoForValidRequestAndSetTokenStoreUsed()
         APIGatewayProxyRequestEvent request = new APIGatewayProxyRequestEvent();
         String validTokenHeader = "Bearer valid-token";
         AccessToken validToken = AccessToken.parse(validTokenHeader, AccessTokenType.BEARER);
-        request.setHeaders(Map.of("Authorization", validTokenHeader));
+        request.setHeaders(Map.of("Authorization", validTokenHeader, SESSION_ID_HEADER, sessionId));
         when(accessTokenService.getAccessTokenFromAuthorizationHeader(any()))
                 .thenReturn(validToken);
         when(userInfoService.populateUserInfo(accessTokenStore)).thenReturn(TEST_SUBJECT_USER_INFO);
+        when(sessionService.getSessionFromRequestHeaders(any()))
+                .thenReturn(Optional.of(testSession));
 
         APIGatewayProxyResponseEvent response = userInfoHandler.userInfoRequestHandler(request);
+        Session updatedSession = testSession.setNewAccount(Session.AccountState.EXISTING);
 
         assertEquals(200, response.getStatusCode());
         assertTrue(
                 response.getBody()
                         .contains(String.format("\"sub\":\"%s\"", TEST_SUBJECT.getValue())));
 
+        verify(sessionService).getSessionFromRequestHeaders(request.getHeaders());
```

Comment:
Do we need this assertion? This feels to me a bit too much like testing the internals of a function. We set up what this should return on line 92, that should be enough.

Reply:
Yeah I see what you mean, feels a bit unnecessary, have removed!